internal/cli/cmd/login/login.go

```diff
@@ -14,46 +14,42 @@
 package login
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"net/url"
-	"os"
 
-	"github.com/go-kit/log/term"
+	"github.com/manifoldco/promptui"
 	persesCMD "github.com/perses/perses/internal/cli/cmd"
 	"github.com/perses/perses/internal/cli/config"
 	"github.com/perses/perses/internal/cli/output"
-	"github.com/perses/perses/internal/cli/read"
 	"github.com/perses/perses/pkg/client/api"
 	"github.com/perses/perses/pkg/client/perseshttp"
+	backendConfig "github.com/perses/perses/pkg/model/api/config"
 	"github.com/spf13/cobra"
-	terminal "golang.org/x/term"
 )
 
-func readPassword() (string, error) {
-	in := os.Stdin
-	if term.IsTerminal(in) {
-		// read using terminal
-		result, err := terminal.ReadPassword(int(in.Fd()))
-		return string(result), err
-	}
-	// read using reader
-	var result string
-	_, err := fmt.Fscan(in, &result)
-	return result, err
-}
+type loginKind int
+
+const (
+	loginKindNative loginKind = iota
+	loginKindOauth
+	loginKindOIDC
+)
 
 type option struct {
 	persesCMD.Option
-	writer       io.Writer
-	url          string
-	username     string
-	password     string
-	accessToken  string
-	refreshToken string
-	insecureTLS  bool
-	apiClient    api.ClientInterface
-	restConfig   perseshttp.RestConfigClient
+	writer           io.Writer
+	url              string
+	username         string
+	password         string
+	loginKind        loginKind
+	selectedProvider string
```

Comment:
for non interactive use cases, we want to be able to select the provider from an argument

Reply:
done